tests/goal/test_goal.py

```diff
@@ -80,6 +80,30 @@ def test_goal_simple_args() -> None:
     verify(result.output)
 
 
+@pytest.mark.usefixtures("proc_mock")
+def test_goal_simple_args_with_input_file() -> None:
+    result = invoke("goal clerk compile approval.teal")
+
+    assert result.exit_code == 0
+    verify(result.output)
+
+
+@pytest.mark.usefixtures("proc_mock")
+def test_goal_simple_args_with_output_file() -> None:
+    result = invoke("goal clerk compile approval.compiled")
```

Comment:
Are you sure this is `with output file` ? I think you need an explicit flag, also compile command migh always expect an input to be provided otherwise it might not know what to compile